src/gh_pr_upsert/cli.py

```diff
@@ -1,13 +1,97 @@
+import sys
 from argparse import ArgumentParser
 from importlib.metadata import version
+from subprocess import CalledProcessError
 
+from gh_pr_upsert import core, git
+from gh_pr_upsert.exceptions import PRUpsertError
 
-def cli(_argv=None):  # pylint:disable=inconsistent-return-statements
-    parser = ArgumentParser()
+
+def cli(_argv=None):  # pylint:disable=too-complex
+    parser = ArgumentParser(description="Create or update a GitHub pull request.")
     parser.add_argument("-v", "--version", action="store_true")
+    parser.add_argument(
+        "--base-remote",
+        help="the git remote to use for the base of the pull request (default: 'origin')",
+        default="origin",
+    )
+    parser.add_argument(
+        "--base-branch",
+        help="the git branch to use for the base of the pull request (default: --base-remote's default branch)",
+    )
+    parser.add_argument(
+        "--local-branch",
+        help="the local git branch to push (default: the current branch)",
+    )
+    parser.add_argument(
+        "--head-remote",
+        help="the git remote to use for the head of the pull request (default: 'origin')",
+        default="origin",
+    )
+    parser.add_argument(
+        "--head-branch",
+        help="the git branch to use for the head of the pull request (default: the branch on --head-remote with same name as --local-branch)",
+    )
+    parser.add_argument(
+        "--title",
+        help="the title to use when creating new pull requests",
+        default="Automated changes by gh-pr-upsert",
+    )
+    parser.add_argument(
+        "--body",
+        help="the body to use when creating new pull requests",
+        default="Automated changes by [gh-pr-upsert](https://github.com/hypothesis/gh-pr-upsert).",
+    )
+    parser.add_argument(
+        "--body-file",
+        help="path to a file containing the body text to use when creating new pull requests",
+    )
+    parser.add_argument(
+        "--close-comment",
+        help="the comment to leave on PRs when closing them",
+        default="It looks like this PR isn't needed anymore, closing it.",
+    )
 
     args = parser.parse_args(_argv)
 
     if args.version:
         print(version("gh-pr-upsert"))
-        return 0
+        sys.exit()
+
+    base_repo = git.GitHubRepo.get(args.base_remote)
+    head_repo = git.GitHubRepo.get(args.head_remote)
+
+    if args.base_branch is None:
+        args.base_branch = base_repo.default_branch
```

Comment:
By default the base branch is the base repo's default branch, usually `main` or `master`